In [30]:
import numpy as np
import os
import pandas as pd
import sys

In [31]:
url = "https://raw.githubusercontent.com/fedhere/PUI2018_fb55/master/Lab4_fb55/times.txt"     

Question: is the new Bus route improving commute?

H0: the commute time is on average the same or longer with the new bus route as it was before: Tnew >= Told

Ha: the commute time is shorter with the new bus route than it was before: Tnew < Told

I set the significance level as: $\alpha=0.05$

In [32]:
bus = pd.read_csv(url, sep=",", names = ["Commute Time"])

In [33]:
bus.head()

Commute Time
0     31.622239
1     32.821376
2     30.229101
3     31.413766
4     39.010550

In [37]:
x = bus['Commute Time'].mean()
y = bus['Commute Time'].std()
N = len(bus)
print ("Mean:", x, "\nStandard Deviation:", y)

Mean: 34.4661616883 
Standard Deviation: 7.13728013375


In [38]:
pmean = 36
pstd = 6

In [39]:
# Calculating Z score 
a = pmean - x 
b = pstd / np.sqrt(N)
zscore= a/b
print(zscore)

2.55639718617


The Z-test score is 2.556, which is higher than the significant level 0.05 as i setted before, thus the null hypothsis can be rejected. It also means that the new bus route is effecient on commuting time. 